probatus/feature_elimination/feature_elimination.py

```diff
@@ -354,19 +369,60 @@ def fit(self, X, y, column_names=None):
         if self.random_state is not None:
             np.random.seed(self.random_state)
 
+        # If to columns_to_keep is not provided, then initialise it by an empty string.
+        # If provided check if all the elements in columns_to_keep are of type string.
+        if columns_to_keep is None:
+            len_columns_to_keep = 0
+        else :
+            if all(isinstance(x,str) for x in columns_to_keep):
+                len_columns_to_keep = len(columns_to_keep)
+            else :
+                raise(ValueError('The current values of columns_to_keep are not allowed.All the elements should be strings.'))
+        
+        # If the column_names parameters is provided, check if the elements in both columns_to_keep & column_names are matching.
+        if column_names is not None and columns_to_keep is not None :
+            if all(x in column_names for x in columns_to_keep):
+                pass
+            else :
+                raise(ValueError('The column names in parameter columns_to_keep and column_names are not macthing.'))
+
+        #Check that the total number of columns to select is less than total number of columns in the data.
+        #only when both parameters are provided.
+        if column_names is not None and columns_to_keep is not None :
+            if (self.min_features_to_select+len_columns_to_keep) > len(self.column_names):
+                raise ValueError('Minimum features to select is greater than number of features.'
+            'Lower the value for min_features_to_select or number of columns in columns_to_keep')
+
+
         self.X , self.column_names = preprocess_data(X, X_name='X', column_names=column_names, verbose=self.verbose)
         self.y = preprocess_labels(y, y_name='y', index=self.X.index, verbose=self.verbose)
         self.cv = check_cv(self.cv, self.y, classifier=is_classifier(self.clf))
 
         remaining_features = current_features_set = self.column_names
         round_number = 0
 
-        while len(current_features_set) > self.min_features_to_select:
+        #Setting up the min_features_to_select parameter.
+        if columns_to_keep is None:
+            pass
+        else:
+            self.min_features_to_select = 0
+            #This ensures that, if columns_to_keep is provided ,the last features remaining are only the columns_to_keep.
+            if self.verbose > 50:
+                print(f'Minimum features to select : {len_columns_to_keep}')
+           
+        #Stop when stopping criteria is met.    
+        stopping_criteria = self.min_features_to_select+len_columns_to_keep
```

Comment:
Why don't we have the stopping_criteria as max of the self.min_features_to_select+len_columns_to_keep and then we don't need part of the logic from above?

Reply:
A max between the two works as well. However we still need to set the min_features_to_select to 0, incase the columns_to_keep is mentioned. However, we can show a warning to the users to let them know that value of min_features_to_select has been over ridden.

Ideally, the min_features_to_select should be set up in the fit method. Now we are trying to set it first and then work around it. Is there scope to change the API now ?

Reply:
Yes, nice idea. Please follow the convention with tthe 
```
if verbose>50: 
    warnings.warn('...')
```

Reply:
Done.